sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -1066,3 +1066,76 @@ test_suite(
     ],
     tests = ["asm_interrupt_handler_fpga_cw310_test_otp_" + otp_name for otp_name in OTP_CFGS_EXEC_DISABLED],
 )
+
+[
+    opentitan_gdb_fpga_cw310_test(
+        name = "rom_e2e_asm_watchdog_fpga_cw310_test_otp_" + otp_name,
+        timeout = "short",
+        gdb_script = """
+            target extended-remote :3333
+
+            echo :::: Send OpenOCD the 'reset halt' command.\\n
+            monitor reset halt
+
+            echo :::: Load ROM symbols into GDB.\\n
+            file rom.elf
+
+            echo :::: Run until we check whether ROM execution is enabled.\\n
+            break _rom_start_boot_maybe_halt
+            continue
+
+            printf ":::: PC=%p. Expected PC=%p.\\n", $pc, _rom_start_boot_maybe_halt
+            if $pc != _rom_start_boot_maybe_halt
+                quit 42
+            end
+
+            echo :::: Pretend execution is enabled.\\n
+            set $pc = _rom_start_boot_exec_en
+
+            echo :::: Run until right after configuring the watchdog timer.\\n
+            break _rom_start_watchdog_enabled
+            continue
+
+            printf ":::: PC=%p. Expected PC=%p.\\n", $pc, _rom_start_watchdog_enabled
+            if $pc != _rom_start_watchdog_enabled
+                quit 43
+            end
+
+            echo :::: Set breakpoint on NMI handler.\\n
+            delete breakpoints
+            break _rom_start_boot
+            break _asm_exception_handler
+            break _rom_interrupt_vector_asm
+            info breakpoints
+
+            echo :::: Wait for interrupt.\\n
+            set $pc = _rom_start_boot_maybe_halt
+            echo :::: Continue.\\n
+            continue
+
+            printf ":::: PC=%p. Expected PC=%p.\\n", $pc, _asm_exception_handler
```

Comment:
@alphan this test passes, but I expected to hit the NMI handler, `_rom_start_boot`! Any guess as to what's happening here?

Reply:
Let's sync offline, please ping me when you want to chat.

Reply:
Per our chat, I was just confused about the value at `_rom_interrupt_vector_asm[31]`. It is `_asm_exception_handler` after all, so the test is working as expected :)

Also:

* I took a stab at using `OT_ADDRESSABLE_LABEL`, but it's not asm-friendly. If the goal is to make it obvious to readers that this is for tests, WDYT about suffixing the labels with `_for_test`?
* I also refactored `rom_start.S` into its own `cc_library`, which should alleviate some pain of duplicating changes for both `:rom` and `:rom_epmp_test`.

Reply:
I think we resolved the first item offline, right? We just want to follow the same convention as `OT_ADDRESSABLE_LABEL()` for the sake of consistency by inlining it here. PLMK if you have any questions.

Thanks for the `rom_start.S` cleanup!

Reply:
> I think we resolved the first item offline, right? We just want to follow the same convention as `OT_ADDRESSABLE_LABEL()` for the sake of consistency by inlining it here. PLMK if you have any questions.

Yep! I defined a `LABEL_FOR_TEST` macro that creates `.local`-scoped labels. I don't feel strongly about the name, just that it has one :P 
 
> Thanks for the `rom_start.S` cleanup!

No problem!